File: lib/features/resources/data/resource_service.dart
import 'package:dio/dio.dart';
import 'package:try_neuro/core/network/http_client.dart';
import 'package:try_neuro/features/resources/domain/resource_model.dart';
import 'package:try_neuro/service_locator.dart';

class ResourceService {
  final Dio _dio = sl<HttpClient>().dio;

  Future<List<Resource>> getResources() async {
    final response = await _dio.get('/resources');
    final List<dynamic> data = response.data;
    return data.map((json) => _fromJson(json)).toList();
  }

  Future<void> addResource({
    required String name,
    String? description,
  }) async {
    await _dio.post('/resources', data: {
      'name': name,
      'description': description,
    });
  }

  Future<void> updateResource(Resource resource) async {
     // Аналогично StaffService, используем POST (save) для обновления, передавая ID
    await _dio.post('/resources', data: {
      'id': resource.id,
      'name': resource.name,
      'description': resource.description,
    });
  }

  Future<void> deleteResource(String resourceId) async {
    await _dio.delete('/resources/$resourceId');
  }

  Resource _fromJson(Map<String, dynamic> json) {
    return Resource(
      id: json['id'],
      name: json['name'],
      description: json['description'],
    );
  }
}

File: frontend-svelte/src/lib/services/resourceService.js
import api from '$lib/api.js';

export const resourceService = {
    // ИЗМЕНЕНО: Добавлен параметр branchId и переход на админский эндпоинт
    async getResources(branchId = null) {
        const response = await api.get('/admin/resources', {
            params: { branchId: branchId }
        });
        return response.data;
    },

    async addResource(data) {
        await api.post('/admin/resources', data);
    },

    async getResourcePhoto(resourceId) {
        const response = await api.get(`/admin/resources/${resourceId}/photo`);
        return response.data;
    },

    async uploadResourcePhoto(resourceId, file) {
        const formData = new FormData();
        formData.append('file', file);
        const response = await api.post(`/admin/resources/${resourceId}/photo`, formData, {
            headers: { 'Content-Type': 'multipart/form-data' }
        });
        return response.data;
    },

    async deleteResourcePhoto(resourceId) {
        const response = await api.delete(`/admin/resources/${resourceId}/photo`);
        return response.data;
    },

    async updateResource(id, data) {
        await api.put(`/admin/resources/${id}`, data);
    },

    async deleteResource(id) {
        await api.delete(`/admin/resources/${id}`);
    },

    async getResourceById(id) {
        // Для деталей используем общий список без фильтра (или можно добавить фильтр позже)
        const resources = await this.getResources();
        return resources.find(r => r.id === id);
    }
};
